app/models/project_group.rb

```diff
@@ -0,0 +1,38 @@
+# == Schema Information
+#
+# Table name: timeline_entries
+#
+#  agent_id          :integer
+#  completed_date    :date
+#  created_at        :datetime         not null
+#  date_change_count :integer          default(0), not null
+#  finalized_at      :datetime
+#  id                :integer          not null, primary key
+#  is_finalized      :boolean
+#  original_date     :date
+#  parent_id         :integer
+#  project_id        :integer
+#  project_type      :string
+#  scheduled_date    :date
+#  step_type_value   :string
+#  type              :string
+#  updated_at        :datetime         not null
+#
+# Indexes
+#
+#  index_timeline_entries_on_agent_id                     (agent_id)
+#  index_timeline_entries_on_project_type_and_project_id  (project_type,project_id)
+#
+# Foreign Keys
+#
+#  fk_rails_a9dc5eceeb  (agent_id => people.id)
+#
+
+require 'chronic'
+class ProjectGroup < TimelineEntry
+  class DestroyWithChildrenError < StandardError; end
+
+  def destroy
+    raise DestroyWithChildrenError.new if children.present?
```

Comment:
Oh yeah! And perhaps add a test for it?

On Tue, Sep 13, 2016 at 10:24 PM, Nathan Lee <notifications@github.com>
wrote:

> In app/models/project_group.rb
> <https://github.com/sassafrastech/madeline_system/pull/136#discussion_r78677596>
> :
>
> > +# Indexes
> > +#
> > +#  index_timeline_entries_on_agent_id                     (agent_id)
> > +#  index_timeline_entries_on_project_type_and_project_id  (project_type,project_id)
> > +#
> > +# Foreign Keys
> > +#
> > +#  fk_rails_a9dc5eceeb  (agent_id => people.id)
> > +#
> > +
> > +require 'chronic'
> > +class ProjectGroup < TimelineEntry
> > +  class DestroyWithChildrenError < StandardError; end
> > +
> > +  def destroy
> > +    raise DestroyWithChildrenError.new if children.present?
>
> Yes, but I am not sure what I was thinking here. It needs to be a callback.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/136/files/d5be92abf5dd5774bf8353cd9021482a60387f44#r78677596>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cPd-CqH3H1ab9IS47LNUKRCVk-GCks5qp1rDgaJpZM4J6NOi>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech